File: examples/cxx/ping_pong/Receiver.cxx
#include "Receiver.hpp"
#include "PingPongTask.hpp"
#include <Host.hpp>
#include <HostNotFoundException.hpp>
#include <Msg.hpp>

#include <Msg.hpp>

MSG_IMPLEMENT_DYNAMIC(Receiver, Process);

const double commSizeLat = 1;
const double commSizeBw = 100000000;

int Receiver::main(int argc, char** argv)
{
	info("Hello");
	double communicationTime=0;
	double time = getClock();
	StringHelper bw;
	
	info("Try to get a task");
	
	PingPongTask* task = reinterpret_cast<PingPongTask*>(Task::get(0));
	
	double timeGot = getClock();
	double timeSent = task->getTime();
	
	delete task;
	
	info(TEXT_("Got at time ") + TEXT_(timeGot));
	info(TEXT_("Was sent at time ") + TEXT_(timeSent));
	
	time = timeSent;
	
	communicationTime = timeGot - time;
	
	info(TEXT_( "Communication time : ") + TEXT_(communicationTime));
	
	info(TEXT_(" --- BW ") + bw.append(commSizeBw/communicationTime,"%07lf") + TEXT_(" ----"));
	
	info("Goodbye!");
	
	delete this;

	return 0;
}

File: examples/java/app/pingpong/Receiver.java
/* Copyright (c) 2006-2014. The SimGrid Team.
 * All rights reserved.                                                     */

/* This program is free software; you can redistribute it and/or modify it
 * under the terms of the license (GNU LGPL) which comes with this package. */

package app.pingpong;
import org.simgrid.msg.Msg;
import org.simgrid.msg.Host;
import org.simgrid.msg.Task;
import org.simgrid.msg.Process;
import org.simgrid.msg.MsgException;
import org.simgrid.msg.NativeException;
import org.simgrid.msg.HostNotFoundException;

public class Receiver extends Process {
  final double commSizeLat = 1;
  final double commSizeBw = 100000000;
  public Receiver(String hostname, String name, String[]args) throws HostNotFoundException, NativeException{
    super(hostname,name,args);
  }

  public void main(String[] args) throws MsgException {
    Msg.info("hello!");
    double communicationTime=0;

    double time = Msg.getClock();

    Msg.info("try to get a task");

    PingPongTask task = (PingPongTask)Task.receive(getHost().getName());
    double timeGot = Msg.getClock();
    double timeSent = task.getTime();

    Msg.info("Got at time "+ timeGot);
    Msg.info("Was sent at time "+timeSent);
    time=timeSent;

    communicationTime=timeGot - time;
    Msg.info("Communication time : " + communicationTime);
    Msg.info(" --- bw "+ commSizeBw/communicationTime + " ----");
    Msg.info("goodbye!");
  }
}